### ISSUE TITLE
Components to create

### ISSUE THREAD

**Issue by jamesplease** [jamesplease/materialish#64]
- [ ] Input
- [ ] Modal ( #3 )
- [ ] ButtonGroup
- [ ] Table
- [ ] Checkbox
- [ ] Toggle

**Comment by conorhastings** [comment]
how do you envision the async autocomplete, consuming a promise and filtering from response?

This one requires a lot of though imo:

- caching
- loading state
- debouncing or throttling
- probably other things

**Comment by jamesplease** [comment]
To be completely honest, I haven't put _any_ thought into what that component will look like yet. We plan to build a "basic" dropdown first, to keep the problem space narrow, and then branch out from there.

**Comment by conorhastings** [comment]
yeah thinks that's the right way to go, something you really want to get right, and with the primitives, async is implementable in userland
